apps/glusterfs/volume_entry_allocate.go

```diff
@@ -262,6 +262,32 @@ func (v *VolumeEntry) prepForBrickReplacement(db wdb.DB,
 	return
 }
 
+func (v *VolumeEntry) generateDeviceFilter(db wdb.RODB) (DeviceFilter, error) {
+
+	var filter DeviceFilter = nil
+	zoneChecking := v.GetZoneCheckingStrategy()
+	if zoneChecking == ZONE_CHECKING_UNSET {
+		zoneChecking = ZoneChecking
+	}
+
+	switch zoneChecking {
+	case ZONE_CHECKING_STRICT:
+		dzm, err := NewDeviceZoneMapFromDb(db)
+		if err != nil {
+			return nil, err
+		}
+
+		filter = dzm.Filter
+	case ZONE_CHECKING_NONE:
+	default:
+		logger.Warning(
+			"ZoneChecking set to unknown value '%v', "+
```

Comment:
To make the log message generic, might skip the word "unknown" .

Reply:
Not sure I understand. This is only the `default:` case. That is exactly for unknown values.